typed-racket-lib/typed-racket/private/type-contract.rkt

```diff
@@ -695,8 +714,25 @@
                                 (hash-set rv v (same any/sc)))))
         (extend-row-constraints vs (list constraints)
           (t->sc body #:recursive-values recursive-values)))
-      ;; FIXME: needs sealing contracts, disabled for now
-      (fail #:reason "cannot generate contract for row polymorphic type")))
+      (match-let ([(PolyRow-names: vs-nm constraints b) type])
+        ;; FIXME: abstract this
+        (define function-type?
+          (let loop ([ty b])
+            (match (resolve ty)
+              [(Function: _) #t]
+              [(Union: elems) (andmap loop elems)]
+              [(Poly: _ body) (loop body)]
+              [(PolyDots: _ body) (loop body)]
+              [_ #f])))
+        (unless function-type?
+          (fail #:reason "cannot generate contract for non-function polymorphic type"))
+        (let ((temporaries (generate-temporaries vs-nm)))
+          (define rv (for/fold ((rv recursive-values)) ((temp temporaries)
+                                                        (v-nm vs-nm))
+                       (hash-set rv v-nm (same (sealing-var/sc temp)))))
+          (parameterize ([bound-names (append (bound-names) vs-nm)])
+            (sealing->/sc temporaries (take constraints 3)
```

Comment:
I should probably add a comment here. I think (but I'm not 100% sure yet) that the constraints for the row variable for augmenting methods are not actually necessary for the contract because it's non-sensical for a class type to have an augment method type without a corresponding public method type. So only the first three sets of constraints are needed.